src/mon/Monitor.cc

```diff
@@ -5337,6 +5337,27 @@ void Monitor::count_metadata(const string& field, Formatter *f)
   f->close_section();
 }
 
+bool Monitor::check_daemon_versions(map<int,std::pair<int, string>> &error){
+  int i = 0;
+  bool warn = false;
+  int start = 0;
+  string first_daemon = "Hello";
+  for (auto& p : mon_metadata) {
+    auto q = p.second.find("ceph_version");
+    string current_daemon = q->second;
+    if (start == 0){
+      first_daemon = current_daemon;
+      start = 1;
+    }else if ((current_daemon != first_daemon) && (start != 0)) {
```

Comment:
It seems to me that 2nd check `(start != 0)` could be avoided as it will always evaluate to `true` after 1st pass.

Reply:
The purpose of that if statement is to get the ceph version of the first node and save it in first_daemon then compare that version to the version of all other nodes. If the versions don't match up or current_daemon != first_daemon then there is an error

Reply:
Your right I completely missed that. Thanks